GeneralsMD/Code/GameEngine/Source/Common/System/BuildAssistant.cpp

```diff
@@ -718,6 +705,19 @@ LegalBuildCode BuildAssistant::isLocationClearOfObjects( const Coord3D *worldPos
 		if (them->isKindOf(KINDOF_INERT))
 			continue;
 
+		if (builderObject && them->getShroudedStatus(builderObject->getControllingPlayer()->getPlayerIndex()) >= OBJECTSHROUD_FOGGED)
+			return LBC_SHROUD;
+
+		//Kris: Patch 1.01 - November 5, 2003
+		//Prevent busy units (black lotus hacking from being moved by trying to place a building -- exploit).
+		if (rel == ALLIES)
+		{
+			if (them->testStatus(OBJECT_STATUS_IS_USING_ABILITY) || them->getAI() && them->getAI()->isBusy())
+			{
+				return LBC_OBJECTS_IN_THE_WAY;
```

Comment:
Side note: I wonder if this code path is missing `TheTerrainVisual->addFactionBib( them, TRUE );`...

Reply:
Separate. Was just wondering about it as I looked at this code here.